game/shared/ehandle.h

```diff
@@ -11,19 +11,30 @@
 #pragma once
 #endif
 
-#include "entityhandle.h"
 #include "entity2/entitysystem.h"
+#include "entityhandle.h"
 
 // -------------------------------------------------------------------------------------------------- //
 // Game-code CBaseHandle implementation.
 // -------------------------------------------------------------------------------------------------- //
 
-inline IHandleEntity* CEntityHandle::Get() const
+inline CEntityInstance* CEntityHandle::Get() const
 {
 	extern CEntitySystem *g_pEntitySystem;
 	return g_pEntitySystem->GetBaseEntity( *this );
 }
 
+// -------------------------------------------------------------------------------------------------- //
+// Game-code CEntityInstance implementation.
+// -------------------------------------------------------------------------------------------------- //
+inline CEntityHandle CEntityInstance::GetRefEHandle() const

```

Comment:
Also no need for that function be in here, define it directly in the CEntityInstance body instead